ulist/python/ulist/io.py

```diff
@@ -1,11 +1,28 @@
 from __future__ import annotations  # To avoid circular import.
 from .ulist import read_csv as _read_csv
-from typing import List, TYPE_CHECKING
+from typing import Dict, TYPE_CHECKING
 
 if TYPE_CHECKING:  # To avoid circular import.
     from . import UltraFastList
 
 
-def read_csv() -> List[UltraFastList]:
+def read_csv(path: str, schema: Dict[str, str]) -> Dict[str, UltraFastList]:
+    """Read the csv file.
+
+    Args:
+        path (str):
+            The path of the csv file.
+        schema (Dict[str,str]):
+            The structure of the csv file, such as
+            `{"foo" : "int", "bar" : "bool"}`
+
+    Returns:
+        Dict[str, UltraFastList]
+    """
     from . import UltraFastList  # To avoid circular import.
-    return [UltraFastList(x) for x in _read_csv()]
+    schema_seq = [x for x in schema.items()]  # To ensure the right order
```

Comment:
Usually, I think people are expecting the order of columns in Python list are the same as the order in the `.csv` file. Shall we also run the tests below:

1. The column in the file is ['foo', 'bar'], the schema is {'foo':  'int', 'bar': 'bool'}, the expected order is ['foo', 'bar']
2. The column in the file is ['bar', 'foo'], the schema is {'foo':  'int', 'bar': 'bool'}, the expected order is ['bar', 'foo']

Reply:
> Usually, I think people are expecting the order of columns in Python list are the same as the order in the `.csv` file. Shall we also run the tests below:
> 
>     1. The column in the file is ['foo', 'bar'], the schema is {'foo':  'int', 'bar': 'bool'}, the expected order is ['foo', 'bar']
> 
>     2. The column in the file is ['bar', 'foo'], the schema is {'foo':  'int', 'bar': 'bool'}, the expected order is ['bar', 'foo']

In fact, `read_csv` will return a dict:
```python
def read_csv(path: str, schema: Dict[str, str]) -> Dict[str, UltraFastList]:
...
```

So the order in `schema_seq` is for the rust code, because:

1. `schema` is a `dict`, so that the fields in `schema` is out of order.
2. The return type of `read_csv` in the rust code is an ordered list, so `schema` can not passed to rust directly as an argument, as python part doesn't know the real order in `.csv` files. And I reorgnize `schema` to `schema_seq` for keeping this order.
3. The return type of `read_csv` in the python code is dict, so actually this order doesn't effect the result  ^_^

So, no matter how the columns arranged in the `.csv` file, the result will always be an unordered `dict`

Reply:
Sorry, I was totally misunderstanding the meaning of "order" here, please just ignore this~.